county_health_preprocessing.ipynb

```diff
@@ -1661,6 +1662,565 @@
+       "state county                                                  \n",
+       "LA    St. Charles Parish                                 11   \n",
+       "MS    Holmes County                                   13.89   \n",
+       "CA    Sacramento County                               10.31   \n",
+       "ND    Adams County                                     9.45   \n",
+       "AR    Hempstead County                                11.93   \n",
+       "\n",
+       "                         Daily fine particulate matter_2008  \\\n",
+       "state county                                                  \n",
+       "LA    St. Charles Parish                              10.54   \n",
+       "MS    Holmes County                                   12.69   \n",
+       "CA    Sacramento County                               11.33   \n",
+       "ND    Adams County                                     6.89   \n",
+       "AR    Hempstead County                                11.37   \n",
+       "\n",
+       "                         Daily fine particulate matter_2009  \\\n",
+       "state county                                                  \n",
+       "LA    St. Charles Parish                              11.74   \n",
+       "MS    Holmes County                                   11.15   \n",
+       "CA    Sacramento County                               10.68   \n",
+       "ND    Adams County                                    11.66   \n",
+       "AR    Hempstead County                                10.59   \n",
+       "\n",
+       "                         Daily fine particulate matter_2010  \\\n",
+       "state county                                                  \n",
+       "LA    St. Charles Parish                              12.46   \n",
+       "MS    Holmes County                                   12.56   \n",
+       "CA    Sacramento County                                8.25   \n",
+       "ND    Adams County                                    10.87   \n",
+       "AR    Hempstead County                                11.36   \n",
+       "\n",
+       "                         Daily fine particulate matter_2011  \n",
+       "state county                                                 \n",
+       "LA    St. Charles Parish                              11.86  \n",
+       "MS    Holmes County                                   12.34  \n",
+       "CA    Sacramento County                                8.68  \n",
+       "ND    Adams County                                     9.79  \n",
+       "AR    Hempstead County                                11.15  \n",
+       "\n",
+       "[5 rows x 93 columns]"
+      ]
+     },
+     "execution_count": 55,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df_train_copy = df_train.copy()\n",
+    "df_train_copy.columns = ['_'.join(col) for col in df_train_copy.columns]\n",
+    "df_train_copy.head()"
```

Comment:
Thanks for reviewing the notebook and making this suggestion. I saw this option when I was researching this part of the notebook. Unfortunately, this doesn't work well because you can't separate the two column values being stitched together easily after you perform this operation. What I needed was to pivot the year_span part of the column multiindex back into a single year_span column, after having done the train/test split with rows for only state and county. Good to know how to do this, but wasn't useful for my particular use case as the `stack` function did exactly what I was looking for. Happy to discuss this further if you'd like.

After giving some of your earlier comments  more thought, I actually think this notebook has too much going on and should be separated into two notebooks instead: 1) for the training/test split and any features we want to extract from shapefiles, etc. and 2) all the work on this health measure dataset as a standalone notebook without any train/test splitting. What do you think about this, @Xuehong-pdx ?

Reply:
Wait, I'm not sure my comment makes much sense as I think this is the old notebook, not the new one. Sorry, about that. The question is still relevant but you won't see the `stack` command used in this earlier version of the notebook